firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/Candidate.kt

```diff
@@ -32,7 +38,58 @@ internal constructor(
   public val safetyRatings: List<SafetyRating>,
   public val citationMetadata: CitationMetadata?,
   public val finishReason: FinishReason?
-)
+) {
+
+  @Serializable
+  internal data class InternalCandidate(
+    val content: Content.InternalContent? = null,
+    val finishReason: FinishReason.InternalFinishReason? = null,
+    val safetyRatings: List<SafetyRating.InternalSafetyRating>? = null,
+    val citationMetadata: CitationMetadata.InternalCitationMetadata? = null,
+    val groundingMetadata: InternalGroundingMetadata? = null,
+  ) {
+    internal fun toPublic(): Candidate {
+      val safetyRatings = safetyRatings?.map { it.toPublic() }.orEmpty()
+      val citations = citationMetadata?.toPublic()
+      val finishReason = finishReason?.toPublic()
+
+      return Candidate(
+        this.content?.toPublic() ?: content("model") {},
+        safetyRatings,
+        citations,
+        finishReason
+      )
+    }
+
+    @Serializable
+    internal data class InternalGroundingMetadata(
+      @SerialName("web_search_queries") val webSearchQueries: List<String>?,
```

Comment:
Since we are already doing this massive work, AFAIK it's not necessary to have the snake case renaming. Can we remove it from the places it doesn't make sense?

Reply:
sorry, what was the resolution here? I couldn't find a comment against it or a code change in favor of.